testing-unit-cs-mocks/Testing.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016-2020, Pulumi Corporation
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Pulumi;
+using Pulumi.Testing;
+
+namespace UnitTesting
+{
+    class Mocks : IMocks
+    {
+        public Task<(string id, object state)> NewResourceAsync(string type, string name, ImmutableDictionary<string, object> inputs, string? provider, string? id)
+        {
+            var outputs = ImmutableDictionary.CreateBuilder<string, object>();
+            
+            // Forward all input parameters as resource outputs, so that we could test them.
+            outputs.AddRange(inputs);
+            
+            // Set the name to resource name if it's not set explicitly in inputs.
+            if (!inputs.ContainsKey("name"))
+                outputs.Add("name", name);
+            
+            if (type == "azure:storage/blob:Blob")
```

Comment:
would this not be better served with a switch? It seems long winded to continually add if statements

Reply:
You are right, but it's are easier to gradually add/explain `if`'s in the blog post. I'll change to switch next time I revisit this example after the blog is published.

Reply:
👍